tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -7127,16 +7116,15 @@ void CGMSHLSLRuntime::EmitHLSLOutParamConversionInit(
         }
 
         llvm::Type *ToTy = tmpArgAddr->getType()->getPointerElementType();
-        Instruction::CastOps castOp =
-            static_cast<Instruction::CastOps>(HLModule::FindCastOp(
-                IsUnsigned(argLV.getType()), IsUnsigned(tmpLV.getType()),
-                outVal->getType(), ToTy));
-
-        Value *castVal = CGF.Builder.CreateCast(castOp, outVal, ToTy);
-        if (!HLMatrixLower::IsMatrixType(ToTy))
-          CGF.Builder.CreateStore(castVal, tmpArgAddr);
-        else
+        if (HLMatrixLower::IsMatrixType(ToTy)) {
+          Value *castVal = CGF.Builder.CreateBitCast(outVal, ToTy);
```

Comment:
I'm not clear on the scenario for which this bitcast is used, though I see that it was doing the same thing by other means before.

Reply:
I agree and also noted it as being highly suspicious. I have on my backlog to write tests for inout conversions.

Reply:
(btw, filed #1890 to remember testing this)